### DISCUSSION TITLE
Persist redux store and page scroll position on back - across ssr pages

### DISCUSSION THREAD

**Discussion by sanishkr** [vercel/next.js#11047] [Help]
Imagine I have a Search result page with first 5 results on ssr next each 5 items are lazy loaded(Infinite Scroll Based on Intersection Observer) from client side, all being pusehd into a redux store.

Now lets say I have clicked on 22nd item after few page scrolls and it takes me to a Product page. When I click mobile native / browser back button, I want it to take me to the same item with same results without rerendering.

[Codesandbox](https://codesandbox.io/s/nextjs-ssr-infinite-scroll-redux-state-back-861sz?fontsize=14&hidenavigation=1&theme=dark)

**Comment by wong2** [comment]
Trying to implement same behavior
